client.go

```diff
@@ -383,7 +383,16 @@ func (c *Client) Delete(ctx context.Context, req Resource, options ...Option) er
 	if err != nil {
 		return err
 	}
-	return c.do(ctx, "DELETE", url, nil, nil)
+	o := &deleteOptions{
+		Kind:              "DeleteOptions",
+		APIVersion:        "v1",
+		PropagationPolicy: "Background",
```

Comment:
This is a change in behaviour: Previously the library used `Orphan` (by not setting a propagation policy). Is this intended? IMHO this would be a major change as it breaks with previous behaviour.